netbout-web/src/test/java/com/netbout/email/EmCatchTest.java

```diff
@@ -0,0 +1,115 @@
+import com.icegreen.greenmail.util.ServerSetup;
+import com.icegreen.greenmail.util.ServerSetupTest;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link EmCatch}.
+ * @author Erim Erturk (erimerturk@gmail.com)
+ * @version $Id$
+ * @since 2.15
+ */
+public final class EmCatchTest {
+
+    /**
+     * EmCatch can periodically read inbox and call Action.run.
+     * @throws Exception If there is some problem inside
+     */
+    @Test
+    public void readsInboxPeriodically() throws Exception {
+        final GreenMail mail = new GreenMail(ServerSetupTest.POP3);
+        mail.start();
+        final ServerSetup setup = mail.getPop3().getServerSetup();
+        final String mailUser = "to";
+        // @checkstyle MagicNumberCheck (1 line)
+        final String from = "from@localhost.com";
+        // @checkstyle LocalFinalVariableNameCheck (1 line)
+        final String to = "to@localhost.com";
+        final String password = "soooosecret";
+        final String subject = GreenMailUtil.random();
+        final String body = GreenMailUtil.random();
+        final GreenMailUser user = mail.setUser(mailUser, password);
+        new EmCatch(
+            // @checkstyle AnonInnerLengthCheck (22 lines)
+            new EmCatch.Action() {
+                @Override
+                public void run(final Message msg) {
+                    MatcherAssert.assertThat(msg, Matchers.notNullValue());
+                    try {
+                        MatcherAssert.assertThat(
+                            msg.getSubject(),
+                            Matchers.equalTo(subject)
+                        );
+                        MatcherAssert.assertThat(
+                            msg.getFrom()[0].toString(),
+                            Matchers.equalTo(from)
+                        );
+                        MatcherAssert.assertThat(
+                            msg.getAllRecipients()[0].toString(),
+                            Matchers.equalTo(to)
+                        );
+                    } catch (final MessagingException ex) {
+                        throw new IllegalStateException(ex);
+                    }
+                }
+            },
+            mailUser,
+            password,
+            setup.getBindAddress(),
+            setup.getPort(),
+            // @checkstyle MagicNumberCheck (1 line)
+            500L
+        );
+        final MimeMessage message = GreenMailUtil.createTextEmail(
+            to,
+            from,
+            subject,
+            body,
+            setup
+        );
+        user.deliver(message);
+        MatcherAssert.assertThat(
+            mail.getReceivedMessages().length,
+            Matchers.equalTo(1)
+        );
+        // @checkstyle MagicNumberCheck (1 line)
+        Thread.sleep(1000);
```

Comment:
@pinaf because mail server have to wait until <code> EmCatch</code> read mail. Without this it will be already stop and i am getting <b> connection refused</b>